diff -dPNur netams-3.3.3a/src/connections.c netams-3.3.3a-new/src/connections.c
--- netams-3.3.3a/src/connections.c	2005-11-20 02:31:02.000000000 +0400
+++ netams-3.3.3a-new/src/connections.c	2006-04-10 03:37:34.000000000 +0500
@@ -92,7 +92,7 @@
 }
 
 #ifdef LINUX
-ssize_t conn_write(void *conn, const char *buf, unsigned i) {
+ssize_t conn_write(void *conn, const char *buf, size_t i) {
 #else 
 int conn_write(void *conn, const char *buf, int i) {
 #endif
diff -dPNur netams-3.3.3a/src/connections.h netams-3.3.3a-new/src/connections.h
--- netams-3.3.3a/src/connections.h	2005-03-02 14:39:30.000000000 +0400
+++ netams-3.3.3a-new/src/connections.h	2006-04-10 03:38:25.000000000 +0500
@@ -13,7 +13,7 @@
 #define CONN_FD_VIRT	-1
 
 #ifdef LINUX
-ssize_t conn_write(void *conn, const char *buf, unsigned i);
+ssize_t conn_write(void *conn, const char *buf, size_t i);
 #else 
 int conn_write(void *conn, const char *buf, int i);
 #endif
